File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        
        box = [] 
        tmp = [] 
        st = [(root, "0") ]
        
        while st: 
            
            node ,path  = st.pop()
            path += str(node.val)
            if not node.left and not node.right: 
                box.append(path)
            
            
            if node.left:
                st.append( [node.left , path])
            if node.right: 
                st.append( [node.right , path ])
               
        ans = 0 
        for b in box: 
            ans += int(b)
        
        return ans 

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0 ; 
    
    public void preorder(TreeNode node , int n){
        if(node !=null ){
            n = n*10 + node.val ; 
            if(node.left == null && node.right ==null){
                ans += n ; 
            }
            preorder(node.left , n ); 
            preorder(node.right , n) ; 
        }
    }
    
    public int sumNumbers(TreeNode root) {
        preorder(root , 0) ;
        return ans ; 
    }
}